cupy/manipulation/add_remove.py

```diff
@@ -13,7 +13,63 @@
 # TODO(okuta): Implement resize
 
 
-# TODO(okuta): Implement trim_zeros
+_first_non_zero_krnl = ReductionKernel(
+    'raw S data, T indices, T len, int64 mode',
+    'T y',
+    '''mode? data[indices] ? indices : len
```

Comment:
`ReductionKernel` unofficially supports `_j` as the index of the inputs. How about using it?